backend/dissemination/workbooklib/end_to_end_core.py

```diff
@@ -43,7 +46,24 @@
 pw.setLevel(logging.INFO)
 
 
+# https://stackoverflow.com/questions/553303/generate-a-random-date-between-two-other-dates
+def random_date(start, end):
+    """
+    This function will return a random datetime between two datetime
+    objects.
+    """
+    delta = end - start
+    int_delta = (delta.days * 24 * 60 * 60) + delta.seconds
+    random_second = randrange(int_delta)  # nosec
+    return start + timedelta(seconds=random_second)
+
+
 def step_through_certifications(sac):
+    d1 = datetime.strptime("1/1/2017 1:30 PM", "%m/%d/%Y %I:%M %p")
+    d2 = datetime.strptime("10/30/2023 4:50 AM", "%m/%d/%Y %I:%M %p")
+    # https://stackoverflow.com/questions/7065164/how-to-make-a-datetime-object-aware-not-naive
+    date = pytz.utc.localize(random_date(d1, d2))
```

Comment:
Just curious why we want random dates now?

Reply:
I'm assuming most of this might go away once we have the historic migrator done.

For testing search, all of the dates were coming in the same: the day the E2E was run. This meant we couldn't actually test sorting, etc.